src/ripple/protocol/LedgerFormats.h

```diff
@@ -86,6 +86,8 @@ enum LedgerEntryType {
 
     ltDEPOSIT_PREAUTH = 'p',
 
+    ltNEGATIVE_UNL = 'N',
```

Comment:
You have some gRPC work to do. You added a new LedgerEntryType, which means we need to add the corresponding LedgerEntryType in protobuf. They are all defined in ledger_objects.proto. You also need to write some C++ code to serialize the ledger object to protobuf. Reach out to me if you need any help.

Reply:
...or you do not create objects on ledger because they might not be necessary and save all that code and work.

I'm somewhat worried that only (nearly) finished implementations are discussed and reviewed here, not feature designs. In my opinion the design of this feature is partially flawed, so I'm not so sure if it makes sense at this stage to discuss about how shared pointers should be used in some places.

Reply:
@cjcobb23 Thanks for the help! 
Implemented with unit tests.